.changeset/ninety-coats-learn.md

```diff
@@ -0,0 +1,7 @@
+---
+'@backstage/core-components': patch
+---
+
+Fix relative `sub-paths` by concatenating the app's base path with them.
+
+As the base path is taken from the configuration file, you must define an implementation for the Config API for the `<Link/>` component to be able to use the `app.baseURL` value defined in `app-config.yaml`.
```

Comment:
Perhaps we can leave this comment out, now that we transparently handle the exception?

The only cases there would be no Config API are in tests and in storybook stories, and in those cases, it's unlikely that callers would rely on this subpath behavior.